pennylane/pulse/rydberg_hamiltonian.py

```diff
@@ -0,0 +1,289 @@
+            qml.evolve(RH)([], t=[0, 10])
+            return qml.expval(qml.PauliZ(0))
+
+    >>> circuit()
+    Array(1., dtype=float32)
+    """
+    if wires is None:
+        wires = list(range(len(register)))
+    elif len(wires) != len(register):
+        raise ValueError("The length of the wires and the register must match.")
+
+    def rydberg_projector(wire: int) -> SProd:
+        """Returns the projector into the Rydberg state for the given wire.
+
+        Args:
+            wire (int): wire value
+
+        Returns:
+            SProd: projector into the Rydberg state
+        """
+        return qml.s_prod(1 / 2, (1 - qml.PauliZ(wire)))
+
+    coeffs = []
+    observables = []
+    for idx, (pos1, wire1) in enumerate(zip(register[:-1], wires[:-1])):
+        for pos2, wire2 in zip(register[(idx + 1) :], wires[(idx + 1) :]):
+            atom_distance = np.linalg.norm(qml.math.array(pos2) - pos1)
+            Vij = interaction_coeff / (abs(atom_distance) ** 6)  # van der Waals potential
+            coeffs.append(Vij)
+            observables.append(qml.prod(rydberg_projector(wire1), rydberg_projector(wire2)))
+
+    return RydbergHamiltonian(
+        coeffs, observables, register=register, interaction_coeff=interaction_coeff
+    )
+
+
+def rydberg_drive(rabi, detuning, phase, wires):
+    r"""Returns a :class:`ParametrizedHamiltonian` representing the action of a driving laser
+    field with the given rabi frequency, detuning and phase acting on the given wires
+
+    .. math::
+
+        \frac{1}{2} \Omega(t) \sum_i (\cos(\phi)\sigma_i^x - \sin(\phi)\sigma_i^y) -
+        \frac{1}{2} \delta(t) \sum_i \sigma_i^z
+
+    where :math:`\Omega`, :math:`\delta` and :math:`\phi` correspond to the rabi frequency, detuning
+    and phase of the laser, :math:`i` correspond to the wire index, and :math:`\sigma^\alpha` for
+    :math:`\alpha = x,y,z` are the Pauli matrices.
+
+    Args:
+        rabi (Union[float, Callable]): float or callable returning the frequency (in MHz) of a laser
+            field
+        detuning (Union[float, Callable]): float or callable returning the detuning (in MHz) of a
+            laser field
+        phase (float): float containing the phase (in radians) of the laser field
+        wires (Union[int, List[int]]): integer or list containing wire values that the laser field
+            acts on
+
+    Returns:
+        RydbergHamiltonian: Hamiltonian representing the action of the laser field on the Rydberg atoms
+
+    **Example**
+
+    .. code-block:: python
+
+        rabi = lambda p, t: p * jnp.sin(jnp.pi * t)
+        detuning = 3 * jnp.pi / 4
+        phase = jnp.pi / 2
+        RH = qml.pulse.rydberg_drive(rabi, detuning, phase, [0, 1, 2, 3])
+
+    >>> RH
+    ParametrizedHamiltonian: terms=2
+
+    .. code-block:: python
+
+        dev = qml.device("default.qubit.jax", wires=4)
+
+        @qml.qnode(dev, interface="jax")
+        def circuit(params):
+            qml.evolve(RH)(params, t=[0, 10])
```

Comment:
Should we note here that this (`qml.evolve(RH)`) is unphysical unless all the atoms are placed so far apart that there is no interaction term, and reference `rydberg_interaction`? Maybe we should also include further up something about 'can be combined with additional drive terms as well as the `rydberg_interaction` term to create a Hamiltonian describing a driven system of Rydberg atoms'

Reply:
Potentially a `..see-also::` block with `ParametrizedHamiltonian`, `ParametrizedEvolution`, `evolve`, and `rydberg_interaction`?

Reply:
I was wondering this because one of the integration tests uses `Hi + Hd` as the hamiltonian that is evolved and was wondering if that should be used. I wanted to keep them separate for the example so that we can show the individual hamiltonians.

Reply:
Very good point 💯 I think it totally makes sense to mention that here. Even though they are two separate functions they are ultimately always linked and the docs should reflect that 🙂

Reply:
I would say a see-also block is good but explicitly mentioning the rydberg interaction term in the example would be good on top 👍